### PR TITLE
Add rules for kernel data structure configs

### PR DESCRIPTION
**Pull request by yuumasato**
#### Description:

Adds rules to check for the following kernel build configs:
- CONFIG_BUG_ON_DATA_CORRUPTION
- CONFIG_DEBUG_CREDENTIALS
- CONFIG_DEBUG_LIST
- CONFIG_DEBUG_NOTIFIERS
- CONFIG_DEGUG_SG

#### Rationale:

- These rules check if the kernel was built with sanity checks for data structures.

### CODE REVIEW

File: linux_os/guide/system/kernel_build_config/kernel_config_debug_list/rule.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+title: 'Enable checks on linked list manipulation'
+
+description: |-
+    Enable this to turn on extended checks in the linked-list walking routines.
+
+    {{{ kernel_build_config_describe_config("CONFIG_DEBUG_LIST", "y") | indent(4) }}}
+
+rationale: |-
+    This add sanity checks to manipulation of linked lists strucutures in the kernel and may
```
Comment (jan-cerny) [inline linux_os/guide/system/kernel_build_config/kernel_config_debug_list/rule.yml:11]:
```suggestion
    This add sanity checks to manipulation of linked lists structures in the kernel and may
```

File: linux_os/guide/system/kernel_build_config/kernel_config_debug_sg/rule.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+identifiers:
+    cce@rhel7: CCE-87147-5
+    cce@rhel8: CCE-87148-3
+    cce@rhel9: CCE-87149-1
+
+ocil_clause: 'the kernel was built with the required value'
+
+ocil: |-
+    {{{ kernel_build_config_ocil("CONFIG_DEGUG_SG", "y") | indent(4) }}}
```
Comment (jan-cerny) [inline linux_os/guide/system/kernel_build_config/kernel_config_debug_sg/rule.yml:27]:
```suggestion
    {{{ kernel_build_config_ocil("CONFIG_DEBUG_SG", "y") | indent(4) }}}
```

File: linux_os/guide/system/kernel_build_config/kernel_config_debug_sg/rule.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+template:
+    name: kernel_build_config
+    vars:
+        config: CONFIG_DEGUG_SG
```
Comment (jan-cerny) [inline linux_os/guide/system/kernel_build_config/kernel_config_debug_sg/rule.yml:32]:
```suggestion
        config: CONFIG_DEBUG_SG
```

File: linux_os/guide/system/kernel_build_config/kernel_config_bug_on_data_corruption/rule.yml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+warnings:
+    {{{ kernel_build_config_warning() | indent(4) }}}
+
+severity: low
+
+identifiers:
+    cce@rhel8: CCE-87304-2
+    cce@rhel9: CCE-87305-9
+
+ocil_clause: 'the kernel was built with the required value'
```
[the remark this answers is not in the document]
Comment (yuumasato) [reply linux_os/guide/system/kernel_build_config/kernel_config_bug_on_data_corruption/rule.yml]: The rule added in this PR are fixed by https://github.com/ComplianceAsCode/content/pull/8483/commits/d1774f5427bf6a9f4a5b242ada09bea2f886c380.I'll fix the previous rules in a separate PR.

File: linux_os/guide/system/kernel_build_config/kernel_config_debug_credentials/rule.yml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    track of the number of pointers from task_structs to any given cred struct, and checks to see
+    that this number never exceeds the usage count of the cred struct.
+
+    Furthermore, if SELinux is enabled, this also checks that the security pointer in the cred
+    struct is never seen to be invalid.
+
+    {{{ kernel_build_config_describe_config("CONFIG_DEBUG_CREDENTIALS", "y") | indent(4) }}}
+
+rationale: |-
+    This adds sanity checks and validations to credential data structrures.
```
Comment (ggbecker) [inline linux_os/guide/system/kernel_build_config/kernel_config_debug_credentials/rule.yml:16]:
```suggestion
    This adds sanity checks and validations to credential data structures.
```

**Review by jan-cerny — APPROVED** [review]

**Pull request merged**